python/lsst/meas/deblender/deblend.py

```diff
@@ -674,50 +677,59 @@ def __init__(self, schema, peakSchema=None, **kwargs):
-                                                   constraints=self.config.constraints,
-                                                   l0_thresh=l0Thresh,
-                                                   l1_thresh=l1Thresh,
-                                                   useStrictMonotonicity=self.config.useStrictMonotonicity,
-                                                   monotonicUseNearest=self.config.useNearestMonotonic,
-                                                   smoothness=self.config.smoothness,
-                                                   slack=self.config.stepSlack,
-                                                   usePsf=self.config.usePsfConvolution,
-                                                   update_order=updateOrder)
+            raise NotImplementedError(err.format(self.config.translationMethod))
+
+        # If the default constraints are not used, set the constraints for
+        # all of the sources
+        constraints = None
+        if (sorted(self.config.constraints) != ['+', '1', 'M', 'S']
+            or ~np.isnan(self.config.l0Thresh)
+            or ~np.isnan(self.config.l1Thresh)
+        ):
+            print("Different constraints")
+            constraintDict = {
+                "+": scarlet.constraints.PositivityConstraint,
+                "1": scarlet.constraints.SimpleConstraint,
+                "M": scarlet.constraints.DirectMonotonicityConstraint(use_nearest=False),
+                "S": scarlet.constraints.DirectSymmetryConstraint(sigma=self.config.symmetryThresh)
+            }
+            for c in self.config.constraints:
+                if constraints is None:
+                    constraints = constraintDict[c]
+                else:
+                    constraints = constraints & constraintDict[c]
+            if constraints is None:
+                constraints = scarlet.constraints.MinimalConstraint()
+            if ~np.isnan(self.config.l0Thresh):
+                constraints = constraints & scarlet.constraints.L0Constraint(self.config.l0Thresh)
+            if ~np.isnan(self.config.l1Thresh):
+                constraints = constraints & scarlet.constraints.L1Constraint(self.config.l1Thresh)
+            if ~np.isnan(self.config.tvxThresh):
+                constraints = constraints & scarlet.constraints.TVxConstraint(self.config.tvxThresh)
+            if ~np.isnan(self.config.tvyThresh):
+                constraints = constraints & scarlet.constraints.TVyConstraint(self.config.tvyThresh)
+
+        multiband_plugin = plugins.DeblenderPlugin(
+            plugins.buildMultibandTemplates,
+            useWeights=self.config.useWeights,
+            usePsf=self.config.usePsfConvolution,
+            constraints=constraints,
+            config=config,
+            maxIter=self.config.maxIter,
+            bg_scale=self.config.bgScale,
+            relErr=self.config.relativeError
+        )
```

Comment:
Why the strange mix between snake case and camel case? and why abbreviate `relativeError`?

Reply:
This was an artifact of using a 3rd party package that adheres to PEP8 naming conventions (`rel_err` is the parameter name in `proxmin` and `scarlet`. I fixed both names in `meas_deblender`.